packages/migration/src/migrations/hearth/20250303113723-create-id-field-for-corrections.ts

```diff
@@ -0,0 +1,353 @@
+      {
+        $addFields: {
+          extensions: {
+            $arrayToObject: {
+              $map: {
+                input: '$composition.section',
+                as: 'el',
+                in: [
+                  {
+                    $let: {
+                      vars: {
+                        firstElement: {
+                          $arrayElemAt: ['$$el.code.coding', 0]
+                        }
+                      },
+                      in: '$$firstElement.code'
+                    }
+                  },
+                  {
+                    $let: {
+                      vars: {
+                        firstElement: { $arrayElemAt: ['$$el.entry', 0] }
+                      },
+                      in: {
+                        $arrayElemAt: [
+                          { $split: ['$$firstElement.reference', '/'] },
+                          1
+                        ]
+                      }
+                    }
+                  }
+                ]
+              }
+            }
+          }
+        }
+      },
+      {
+        $lookup: {
+          from: 'Patient',
+          localField: 'extensions.child-details',
+          foreignField: 'id',
+          as: 'child'
+        }
+      },
+      { $unwind: '$child' },
+      {
+        $lookup: {
+          from: 'Location',
+          localField: 'extensionsObject.regLastOffice',
+          foreignField: 'id',
+          as: 'office'
+        }
+      },
+      { $unwind: '$office' },
+      {
+        $addFields: {
+          'office.district': {
+            $arrayElemAt: [{ $split: ['$office.partOf.reference', '/'] }, 1]
+          }
+        }
+      },
+      {
+        $lookup: {
+          from: 'Location',
+          localField: 'office.district',
+          foreignField: 'id',
+          as: 'district'
+        }
+      },
+      { $unwind: '$district' },
+      {
+        $addFields: {
+          'district.state': {
+            $arrayElemAt: [{ $split: ['$district.partOf.reference', '/'] }, 1]
+          }
+        }
+      },
+      {
+        $lookup: {
+          from: 'Location',
+          localField: 'district.state',
+          foreignField: 'id',
+          as: 'state'
+        }
+      },
+      { $unwind: '$state' },
+      {
+        $project: {
+          _id: 0,
+          id: '$meta.versionId',
+          gender: '$child.gender',
+          reason: '$reason.text',
+          extensions: '$extensions',
+          officeName: '$office.name',
+          districtName: '$district.name',
+          stateName: '$state.name',
+          event: 'Birth',
+          createdAt: {
+            $dateFromString: { dateString: '$lastModified' }
+          }
+        }
+      },
+      {
+        $out: {
+          db: 'performance',
+          coll: 'corrections'
+        }
+      }
+    ])
+    .toArray()
+
+  await dashboardDb
+    .collection('corrections')
+    .createIndex({ id: 1 }, { unique: true })
+}
```

Comment:
For a field to be usable in the `$merge` pipeline, there needs to be a unique index for it